lib/recording/RecordingManager.js

```diff
@@ -50,133 +50,83 @@ class RecordingManager {
+            this.recordingList[entryId].lastUpdateTime = new Date();
+            return Q.resolve({session:this.recordingList[entryId], started:true});
+        }
+    }
 
+    startRecording(entryObj, mp4FilesQueue = null) {
+        return Q.resolve().then(()=> {
             let startRecordingTime = new Date();
             logger.debug(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Start recording`);
-            return getRecordingEntrySession.call(this, entryObj)
-                .then((recordingEntrySession)=> {
-                    if (recordingEntrySession == null) {
-                        return Q.resolve();
-                    }
-                    let startSessionPromise;
-                    return qio.exists(recordingEntrySession.recordingSessionPath).then((sessionExist)=> {
-                        if (sessionExist) {
-                            startSessionPromise = startExistSession(recordingEntrySession);
-                        }
-                        else {
-                            startSessionPromise = startNewSession(recordingEntrySession);
-                        }
-                        return startSessionPromise
-                            .then(()=> {
-                                // Once start process is complete, release lock on files queue
-                                recordingEntrySession.mp4FilesQueueLock = false;
-                                let startRecordingTimeDuration = new Date() - startRecordingTime;
-                                logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Start recording is completed, time : ${startRecordingTimeDuration} ms`);
-                            })
-                    });
-                })
-                .catch((err)=> {
-                    // In case of error, pass message on and print to log
-                    return Q.reject(err);
+            return this.getRecordingEntrySession(entryObj, mp4FilesQueue)
+                .then((recordingObj)=> {
+                    return recordingObj.session.verifyFlavorSetMatchPlaylistFlavors(entryObj)
+                        .then((flavorsMatch) => {
+                            if (!flavorsMatch) {
+                                return recordingObj.session.splitRecordedEntry(entryObj)
+                                    .then(() => {
+                                        logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] after split, [${recordingObj.session.recordedEntryId}] replaced by [${entryObj.recordedEntryId}]`);
+                                        return this.getRecordingEntrySession(entryObj);
+                                    })
+                            } else {
+                                return recordingObj;
+                            }
+                        })
+                        .then((recordingObj) => {
+                            if (!recordingObj.started) {
```

Comment:
In the past Ron returned null instead of recordedEntrySession and that was the indicator that a recording session exists. I replaced it with the isStarted flag